server.js

```diff
@@ -13,6 +17,96 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
+app.post("/bookings", function (request, response) {
+  
+  let id = bookings.length;
+  const {
+    roomId,
+    title,
+    firstName,
+    surname,
+    email,
+    checkInDate,
+    checkOutDate,
+  } = request.body;
+  let newBooking = {
+    id,
+    roomId,
+    title,
+    firstName,
+    surname,
+    email,
+    checkInDate,
+    checkOutDate,
+  };
+  bookings.push(newBooking);
+  response.send(bookings);
+  if (
+    newBooking.roomId &&
+    newBooking.roomId > 0 &&
+    newBooking.title &&
+    newBooking.title.length > 0 &&
+    newBooking.firstName &&
+    newBooking.firstName.length > 0 &&
+    newBooking.surname &&
+    newBooking.surname.length > 0 &&
+    newBooking.email &&
+    newBooking.email.length > 0 &&
+    newBooking.checkInDate &&
+    newBooking.checkInDate.length > 0 &&
+    newBooking.checkOutDate &&
+    checkOutDate.length > 0
+  ) {
+    bookings.push(newBooking);
+    response.status(201).send(bookings);
+  } else response.status(400).send(` check The above has keys have values.`);
+
+});
+
+app.get("/bookings/:id", function (request, response) {
+  let booking = bookings.find((book) => book.id === Number(request.params.id));
+  booking
+    ? response.status(200).send(booking)
+    : response.status(404).send("Sorry, booking not found");
+});
+
+app.delete("/bookings/:id", function (request, response) {
+  let bookingToDelete = bookings.find(
+    (el) => el.id === Number(request.params.id)
+  );
+  bookings.splice(bookings.indexOf(bookingToDelete), 1);
+  response.send(bookings);
+});
+app.get("/bookings/search", function (request, response) {
+  let searchedDate = request.query.date;
+  const elementToFind = bookings.find(
+    (book) => book.checkInDate === searchedDate || book.checkOutDate === searchedDate
+  );
+  response.send(elementToFind);
+});
+
+async function isEmailValid(email) {
+  return emailValidator.validate(email);
+}
```

Comment:
Nice use of an external library here! A preference thing though, is that 
```
async function isEmailValid(email) {
   return emailValidator.validate(email)
}
```

is the exact same as 
```
async function isEmailValid = emailValidator.validate
```
and the `emailValidator.validate` is already giving the information that "this function validates an email", so I think creating the `isEmailValid` function here is unecessary in this case.

It's a nice pattern to have though, and it's certainly a nit-pick as the `is...` function convention for checking booleans is quite useful in large codebases.